test/source/tests/settings.ts

```diff
@@ -970,6 +970,25 @@ export const defineSettingsTests = (testVariant: TestVariant, testWithBrowser: T
       await settingsPage.close();
     }));
 
+    ava.default('settings - ensure gracious behavior & ui should remain functional when updating client configuration', testWithBrowser(undefined, async (t, browser) => {
+      const acct = 'settings@settings.flowcrypt.test';
+      mockBackendData.clientConfigurationByAcctEmail[acct] = keyManagerAutogenRules;
+      const setupPage = await BrowserRecipe.openSettingsLoginApprove(t, browser, acct);
+      await SetupPageRecipe.autoSetupWithEKM(setupPage);
+      // open the settings page
+      const accessToken = await BrowserRecipe.getGoogleAccessToken(setupPage, acct);
+      await setupPage.close();
+      // Set invalid client configuration and check if it ensures gracious behavior & ui remain functional
+      mockBackendData.clientConfigurationByAcctEmail[acct] = {
+        // flags is required but don't return it (to mock invalid client configuration)
+        key_manager_url: 'https://localhost:8001/flowcrypt-email-key-manager'
+      };
+      const extraAuthHeaders = { Authorization: `Bearer ${accessToken}` };
+      // const gmailPage =
+      const gmailPage = await browser.newPage(t, TestUrls.mockGmailUrl(), undefined, extraAuthHeaders);
+      await PageRecipe.waitForToastToAppearAndDisappear(gmailPage, 'Failed to update FlowCrypt Client Configuration: Missing client configuration flags.');
+    }));
```

Comment:
It'd be good to load the client configuration `cryptup_<acct>_rules` from local storage to check that it's unchanged the way we do in `settings - client configuration gets updated on settings and content script reloads` test.
You can do using the same `setupPage` (setupPage.getFromLocalStorage), I guess (no need to close it).
And the client configuration is also updated when opening `settings` page, need to check its behaviour too.

Reply:
Yeah, nice suggestion